### PR TITLE
cake 5: Add native type hints for I18n

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

**Comment by othercorey** [comment]
The only type changes I made were adding `int` to the Number formatting functions.

File: src/I18n/Date.php [hunk 1]
```diff
@@ -131,8 +134,10 @@ class Date extends MutableDate implements I18nDateTimeInterface
      * @param \DateTimeZone|string|null $tz The timezone in which the date is taken.
      *                                  Ignored if `$time` is a DateTimeInterface instance.
      */
-    public function __construct($time = 'now', $tz = null)
-    {
+    public function __construct(
+        DateTime|DateTimeImmutable|string|int|null $time = 'now',
+        DateTimeZone|string|null $tz = null
+    ) {
         deprecationWarning(
```
Comment (ADmad) [inline src/I18n/Date.php:141]: I thought I cleaned up these deprecation. Probably I have them locally and forgot to make a PR.

Comment (ADmad) [reply src/I18n/Date.php:141]: Actually it's because https://github.com/cakephp/cakephp/pull/15744 hasn't been merged yet and now again has conflicts 😞.

Comment (othercorey) [reply src/I18n/Date.php:141]: I would wait for the View PR to merge as well then we can resolve this.

**Pull request merged**